### PR TITLE
Check type before using them like obj/array

### PR DESCRIPTION
**Pull request by iproctor**

### CODE REVIEW

**Comment by Raynos** [comment]
lgtm lol.

File: index.js [hunk 1]
```diff
@@ -1,6 +1,10 @@
 module.exports = filterObjectOnSchema;
 
 
+function isObject(obj) {
+    return obj === Object(obj);
```
Comment (kriskowal) [inline index.js:5]: technically true, but potentially churns an object, obj && typeof obj === 'object' is equivalent.

**Comment by iproctor** [comment]
Hmm
I was afraid it could be an array

> obj = []
> []
> obj && typeof obj === 'object'
> true

**Comment by iproctor** [comment]
Though I guess it wouldn't crash even if were an array

**Comment by kriskowal** [comment]
Object(obj) === obj is true for arrays as well. You’ll need to `&& !Array.isArray(obj)`

**Pull request merged**